test/integration/production/pages/counter.js

```diff
@@ -0,0 +1,31 @@
+import Link from 'next/link'
+import { Component } from 'react'
+import Router from 'next/router'
+
+let counter = 0
+
+export default class extends Component {
+  increase () {
+    counter++
+    this.forceUpdate()
+  }
+
+  visitQueryStringPage () {
+    const href = { pathname: '/nav/querystring', query: { id: 10 } }
+    const as = { pathname: '/nav/querystring/10', hash: '10' }
+    Router.push(href, as)
+  }
+
+  render () {
+    return (
+      <div id='counter-page'>
+        <Link href='/no-such-page'><a id='no-such-page'>No Such Page</a></Link>
```

Comment:
Can you also add a case with `prefetch` 🙏

Reply:
prefetch doesn't do reload the page (And we can't). It has no effect.